### ISSUE TITLE
Is clustered-object-pool thread safe ?

### ISSUE THREAD

**Issue by ado2000** [bbottema/clustered-object-pool#2]
Is clustered-object-pool thread safe ?

Seems not,

this is my error, m'i doing something wrong ?

```

Call this code in a thread:
PoolableObject<Socket> poolableSocket = pool.claimResourceFromPool(new ResourceKey.ResourceClusterAndPoolKey(fromIP, domain));
					
Socket socket = poolableSocket.getAllocatedObject();

java.util.ConcurrentModificationException
	at java.util.LinkedList$ListItr.checkForComodification(LinkedList.java:966)
	at java.util.LinkedList$ListItr.next(LinkedList.java:888)
	at org.bbottema.clusteredobjectpool.core.ResourcePools.findResourcePool(ResourcePools.java:76)
	at org.bbottema.clusteredobjectpool.core.ResourcePools.claimResource(ResourcePools.java:67)
	at org.bbottema.clusteredobjectpool.core.ResourceClusters.claimResourceFromPool(ResourceClusters.java:138)
```

Thanks

**Comment by stefanocke** [comment]
We also got that exception. It was (sometimes) reproducable in a unit test with high concurrency.
It seems that **org.bbottema.clusteredobjectpool.core.ResourcePools.clusterCollection** is not thread safe. 

Since we do not call register registerResourcePool in advance, I assume, the following hapens:

- Thread 1 calls registerResourcePool in claimResourceFromPool
- Thread 2 iterates through clusterCollection and detects the concurrent modification by thread 1

However, an easy workaround (for us) is to call registerResourcePool() in advance.

**Comment by bbottema** [comment]
Fixed by #4. Released in 2.1.0.
